### PR TITLE
Fix postsubmit rerun based on checkrun

### PR DESCRIPTION
**Pull request by keyonghan**
Fixes: https://github.com/flutter/flutter/issues/147007

### CODE REVIEW

**Comment by ricardoamador** [comment]
Just a heads up that this will conflict with the buildbucket changes. If we can, can we wait for those changes to land? I can help convert this afterwards if you like.

File: app_dart/lib/src/service/luci_build_service.dart [hunk 1]
```diff
@@ -407,12 +410,40 @@ class LuciBuildService {
...
+      );
+    } catch (error) {
+      log.severe(
+        'updating task ${taskDocument.taskName} of commit ${taskDocument.commitSha} failure: $error. Skipping rescheduling.',
+      );
+      return;
+    }
+    tags['current_attempt'] = <String>[(taskDocument.attempts!).toString()];
+    log.info('Updated input ${build.input!} tags $tags');
+    final BatchRequest request = BatchRequest(
```
Comment (ricardoamador) [inline app_dart/lib/src/service/luci_build_service.dart:432]: I guess I am not understanding, is this really a postsubmit task that is run in presubmit for packages?

Comment (keyonghan) [reply app_dart/lib/src/service/luci_build_service.dart:432]: Postsubmit task is for postsubmit CI. It shares the same github UI for checkrun view. You can view the postsubmit ones:
Screenshot 2024-04-22 at 3 55 07 PM

**Comment by keyonghan** [comment]
Validated this works with https://github.com/flutter/packages/runs/24162936475.
Manually reran https://ci.chromium.org/ui/p/flutter/builders/prod/Linux_android%20android_platform_tests_shard_6%20stable/1110/infra, and it successfully updated database and the GitHub checkrun.

**Review by ricardoamador — APPROVED** [review]
LGTM

**Pull request closed without merging**